src/folding/hypernova/nimfs.rs

```diff
@@ -282,10 +290,13 @@ impl<C: CurveGroup> NIMFS<C> {
         }
 
         // Step 1: Get some challenges
-        let gamma: C::ScalarField = transcript.get_and_append_challenge(b"gamma").unwrap();
-        let beta: Vec<C::ScalarField> = transcript
-            .get_and_append_challenge_vectors(b"beta", ccs.s)
-            .unwrap();
+        let gamma_scalar = C::ScalarField::from_le_bytes_mod_order(b"gamma");
+        transcript.absorb(&gamma_scalar);
+        let gamma: C::ScalarField = transcript.get_challenge();
```

Comment:
I'm not an expert on transcripts, so not sure about this, but if these strings are not essential maybe we could skip them so in-circuit don't add extra constraints?

The reasoning would be that we are already absorbing the value of the computed challenge, and we're interested into absorbing the values computed through the protocol to prevent the prover from 'choosing' the 'random challenges' obtainted, but some hardcoded strings would not affect that, and furthermore, later when we reproduce the transcript in-circuit, absorbing strings would add extra constraints (although a small amount).
Maybe the 'absorbing the string' tradition comes from Merlin (https://merlin.cool , used by the transcript from the Espresso impl), which allows to use strings as kind of separators when absorbing values into the transcript.
I've skimmed through the Merlin docs but didn't find a reason for absorbing the strings, but again, I'm not sure about this, so posting this as a question for others here. (tagging @CPerezz and @han0110 who might have more insight here)

Reply:
I think it's like a domain separator for the underlying hash function, [`zcash/halo2` does the same thing](https://github.com/zcash/halo2/blob/7fd2ce259ec3d0b0e3ede3fa20e4cdcfc395efc9/halo2_proofs/src/transcript.rs#L122), so I guess it'd be a good practice.

But I also don't understand the theoretical part, and not sure where convention this comes from.